packages/core/src/interpreter.ts

```diff
@@ -484,6 +490,12 @@ export class Interpreter<
     };
   }
 
+  public getPersistedState(): PersistedFrom<TBehavior> | undefined {
```

Comment:
When this can return `undefined` if the snapshot is always available synchronously right after creation?

Reply:
When `.getPersistedState` is not defined

Reply:
It would be great to have a test case for this. I wonder though... we have quite a bit of different concepts here:
- snapshots -> public~ state
- internal state
- persisted state (a serializable internal state)

It's quite a lot, I wonder if we could simplify this somehow. Perhaps serializing could be an enhancement and if there is no `getPersistedState` then maybe we should just return the internal state